### PR TITLE
Adding Okta OAuth2 Example to Docs

### PR DESCRIPTION
**Pull request by nniehoff**
This PR adds documentation for configuring social auth with Okta using OAuth2 instead of OIDC which may be necessary for some deployments of Okta.  Also included is an example of syncing groups claims from Okta with Nautobot groups.

### CODE REVIEW

File: nautobot/docs/configuration/authentication/sso.md [hunk 1]
```diff
@@ -135,7 +154,23 @@ SOCIAL_AUTH_OKTA_OPENIDCONNECT_SECRET = '<Client Secret From Okta>'
...
+SOCIAL_AUTH_OKTA_OAUTH2_SCOPE = ['groups']
+```
+
+for OpenID:
+
+```python
+SOCIAL_AUTH_OKTA_OPENIDCONNECT_SCOPE = ['groups']
+```
+
+In order to use this returned scope a custom function needs to be written and added to the `SOCIAL_AUTH_PIPELINE` as described in the [python social auth core pipeline](https://python-social-auth.readthedocs.io/en/stable/pipeline.html) documentation.  An example to sync groups with Okta is provided in the [`/examples/okta`](https://github.com/nautobot/nautobot/tree/develop/examples/okta) folder.
```
Comment (jathanism) [inline nautobot/docs/configuration/authentication/sso.md:173]:
```suggestion
In order to use this returned scope a custom function needs to be written and added to the `SOCIAL_AUTH_PIPELINE` as described in the [`python-social-auth` authentication pipeline documentation](https://python-social-auth.readthedocs.io/en/stable/pipeline.html).

An example to sync groups with Okta is provided in the [`examples/okta`](https://github.com/nautobot/nautobot/tree/develop/examples/okta) folder in the root of the Nautobot repository.
```

**Review by jathanism — APPROVED** [review]
This is fantastic! Please just accept my suggestions and we can ship this. :)

File: examples/okta/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "social_core.pipeline.user.get_username",
+    "social_core.pipeline.user.create_user",
+    "social_core.pipeline.social_auth.associate_user",
+    "social_core.pipeline.social_auth.load_extra_data",
+    "social_core.pipeline.user.user_details",
+    "my_custom_module.okta.group_sync",
+)
+```
+
+If the name of the claim from okta is different both the `SOCIAL_AUTH_OKTA_OAUTH2_SCOPE` and the script will need to be modified.  This example is provided as is and may very well be unique for your okta deployment.
```
Comment (glennmatthews) [inline examples/okta/README.md:26]:
```suggestion
If the name of the claim from Okta is different both the `SOCIAL_AUTH_OKTA_OAUTH2_SCOPE` and the script will need to be modified.  This example is provided "as is" and may very well be unique for your Okta deployment.
```

File: examples/okta/group_sync.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+"""Additional functions to process an OKTA user."""
```
Comment (glennmatthews) [inline examples/okta/group_sync.py:1]:
```suggestion
"""Additional functions to process an Okta user."""
```

File: nautobot/docs/configuration/authentication/sso.md [hunk 4]
```diff
@@ -122,7 +122,26 @@ The following guides are provided for some of the most common authentication met
     * *Login redirect URIs*: should be the Base URI plus `/complete/okta-openidconnect/` such as `https://nautobot.example.com/complete/okta-openidconnect/`
     * *Logout redirect URIs*: should be the Base URI plus `/disconnect/okta-openidconnect/` such as `https://nautobot.example.com/disconnect/okta-openidconnect/`
 
-3. Once the application is configured in Okta, edit your `nautobot_config.py` as follows:
+3. Once the application is configured in Okta, SSO can either be configured with OAuth2 or OpenID Connect (OIDC).  When using the organizations authentication server OAuth2 is preferred, with custom Okta authentication backends use OIDC.
```
Comment (glennmatthews) [inline nautobot/docs/configuration/authentication/sso.md:125]:
```suggestion
3. Once the application is configured in Okta, SSO can either be configured with OAuth2 or OpenID Connect (OIDC).  When using an organization's authentication server OAuth2 is preferred; with custom Okta authentication backends, use OIDC.
```

**Review by glennmatthews — APPROVED** [review]
Great addition!

**Pull request merged**